rosidl_typesupport_opensplice_cpp/CMakeLists.txt

```diff
@@ -23,6 +27,17 @@ ament_export_include_directories(include include/${PROJECT_NAME}/impl)
 
 ament_python_install_package(${PROJECT_NAME})
 
+add_library(${PROJECT_NAME} SHARED src/identifier.cpp)
```

Comment:
I don't think it should be in `rmw_opensplice_cpp`. Moving the definition of the symbol to the rmw package would imply that a message package builds a library which contains a declared symbol which is not defined in any of its dependencies.

When the definition is coming from the rmw package I would call that a dependency for each message package. And we don't want to introduce a dependency in that direction.